### PR TITLE
Better error handling for watching a list of resources specified in a file

### LINKED ISSUE #15665: Watching a list of resource doesn't work
**Problem**:
`kubectl get -f <file> --watch` won't work if the file contains a list (api.List, api.ReplicationControllerList, etc). 

For example, if we want to watch a ReplicationControllerList with 2 rc:
```console
$ kubectl get -f hack/testdata/multi-resource-rclist.json -w
CONTROLLER   CONTAINER(S)     IMAGE(S)           SELECTOR    REPLICAS   AGE
mock         mock-container   kubernetes/pause   app=mock    1          <unknown>
mock2        mock-container   kubernetes/pause   app=mock2   1          <unknown>
error: resource name may not be empty
```
Note that watch will first print the object being watched and then start watching. The object printed here should be but is *not* retrieved from server, because the list doesn't have a "name" and therefore the code failed to refresh the object. 

**Solution**:
Since we don't support watching multiple resources, we should print an error message when we find the file contains a list. The above example should print:
```console
$ kubectl get -f hack/testdata/multi-resource-rclist.json -w
error: watch is only supported on a single resource, but a list of resources is found
```

cc @bgrant0607 @smarterclayton

### PR DESCRIPTION
**Pull request by janetkuo**
Fixes #15665

cc @bgrant0607 @smarterclayton

### CODE REVIEW

File: pkg/kubectl/resource/visitor.go [hunk 1]
```diff
@@ -564,6 +564,9 @@ func RetrieveLatest(info *Info, err error) error {
 	if err != nil {
 		return err
 	}
+	if runtime.IsListType(info.Object) {
+		return fmt.Errorf("watch is only supported on a single resource, but a list of resources is found")
```
Comment (bgrant0607) [inline pkg/kubectl/resource/visitor.go]: This error is misleading given:
https://github.com/kubernetes/kubernetes/issues/15665#issuecomment-148216664

how about:
on individual resources and resource collections

**Comment by JanetKuo** [comment]
Comment addresses. PTAL

**Comment by smarterclayton** [comment]
LGTM

**Comment by wojtek-t** [comment]
Jenkins test flake:
"Kubectl client Guestbook application should create and stop a working application [Conformance]"

@k8s-bot test this please

**Pull request merged**